catalog/views.py

```diff
@@ -1,3 +1,137 @@
 from django.shortcuts import render
+from .models import Book, Author, BookInstance, Genre
+from django.views import generic
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.contrib.auth.mixins import PermissionRequiredMixin
 
 # Create your views here.
+def index(request):
+    """View function for home page of site."""
+
+    # Generate counts of some of the main objects
+    num_books = Book.objects.all().count()
```

Comment:
nếu ở sau có query thì có thể bỏ all ko cần thiết nhé
```suggestion
    num_books = Book.objects.count()
```